### PR TITLE
retinanet onnx model has Upsample error

### PR DESCRIPTION
**Pull request by vgavrilo**
retinanet onnx model has Upsample error

Ticket: #31337

Code:

* [x] Comments
* [x] Code style (PEP8)
* [x] Transformation generates reshape-able IR: yes
* [x] Transformation preserves node names: yes

Validation: 

* [x] Unit tests
* [x] Framework layer tests:  N/A because there are no new supported layers
* [x] Transformation tests: N/A because there are no new transformations
* [x] e2e model test: N/A because there is no new model
* [x] MO IR Reader tests: N/A because there is no new model

Documentation:

* [x] Supported frameworks layers list: N/A because there are no new operations
* [x] Supported **public** models list: N/A because there is no new **public** model
* [x] New operations specification: N/A because there are no new operations
* [x] Guide on how to convert the **public** model: N/A because there is no new **public** model
* [x] User guide update: N/A because there is no need to update common user guide

Other:

* [x] Sample/Demo application to infer the model: N/A since the model can be executed using existing classification sample

### CODE REVIEW

File: model-optimizer/extensions/ops/ONNXResize11.py [hunk 1]
```diff
@@ -61,12 +61,14 @@ def onnx_resize_infer(node: Node):
             assert input2_value is not None, \
                 "Node {} with op {} has no value in input port 2".format(node.soft_get('name', node.id), node.op)
             scale = np.array(input2_value)
-            output_shape = np.floor(input_shape * scale + 1.0e-6).astype(np.int64)
+            output_shape = np.floor((input_shape + 1.0e-6) * scale).astype(np.int64)
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/ONNXResize11.py:64]: Why did you change this? Have you seen such a formula in the ONNX runtime?

File: model-optimizer/extensions/ops/ONNXResize11.py [hunk 2]
```diff
@@ -61,12 +61,14 @@ def onnx_resize_infer(node: Node):
...
         else:
             # i.e. input 'sizes' is given
             sizes = node.in_port(3).data.get_value()
             assert sizes is not None, \
                 "Node {} with op {} has no value in input port 3".format(node.name, node.op)
-            output_shape = int64_array(sizes)
+            output_shape = input_shape.copy()
+            indices = np.array(range(2, len(input_shape))) # indices of spatial dimensions
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/ONNXResize11.py:71]: Please, rename the variable to `spatial_dimension_indices`. BTW, why did you converted it to `np.array`? Is it really needed?

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/extensions/ops/ONNXResize11.py (hunk 2 above)
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/ONNXResize11.py:71]: Please, add unit tests for shape calculation for the case with `scales` and `sizes`

File: model-optimizer/extensions/ops/ONNXResize11.py (hunk 1 above)
Comment (vgavrilo) [reply model-optimizer/extensions/ops/ONNXResize11.py]: Fixed.

File: model-optimizer/extensions/ops/ONNXResize11.py (hunk 2 above)
Comment (vgavrilo) [reply model-optimizer/extensions/ops/ONNXResize11.py]: Fixed.

Comment (vgavrilo) [reply model-optimizer/extensions/ops/ONNXResize11.py]: Done.

**Review by lazarevevgeny — APPROVED** [review]
